### ISSUE TITLE
Error when unioning signal domains

### ISSUE THREAD

**Issue by domoritz** [vega/vega#1105]
```json
{
  "$schema": "https://vega.github.io/schema/vega/v3.0.json",
  "data": [
    {
      "name": "source_0",
      "url": "data/cars.json",
      "format": {"type": "json"}
    },
    {
      "name": "data_0",
      "source": "source_0",
      "transform": [
        {
          "type": "formula",
          "expr": "toNumber(datum[\"Miles_per_Gallon\"])",
          "as": "Miles_per_Gallon"
        },
        {
          "type": "extent",
          "field": "Miles_per_Gallon",
          "signal": "child_Miles_per_Gallon_bin_maxbins_6_Miles_per_Gallon_extent"
        },
        {
          "type": "bin",
          "field": "Miles_per_Gallon",
          "as": [
            "bin_maxbins_6_Miles_per_Gallon",
            "bin_maxbins_6_Miles_per_Gallon_end"
          ],
          "signal": "child_Miles_per_Gallon_bin_maxbins_6_Miles_per_Gallon_bins",
          "maxbins": 6,
          "extent": {
            "signal": "child_Miles_per_Gallon_bin_maxbins_6_Miles_per_Gallon_extent"
          }
        }
      ]
    },
    {
      "name": "data_1",
      "source": "source_0",
      "transform": [
        {
          "type": "formula",
          "expr": "toNumber(datum[\"Horsepower\"])",
          "as": "Horsepower"
        },
        {
          "type": "extent",
          "field": "Horsepower",
          "signal": "child_Horsepower_bin_maxbins_6_Horsepower_extent"
        },
        {
          "type": "bin",
          "field": "Horsepower",
          "as": [
            "bin_maxbins_6_Horsepower",
            "bin_maxbins_6_Horsepower_end"
          ],
          "signal": "child_Horsepower_bin_maxbins_6_Horsepower_bins",
          "maxbins": 6,
          "extent": {
            "signal": "child_Horsepower_bin_maxbins_6_Horsepower_extent"
          }
        }
      ]
    }
  ],
  "scales": [
    {
      "name": "color",
      "type": "bin-ordinal",
      "domain": {
        "fields": [
          {
            "signal": "sequence(child_Miles_per_Gallon_bin_maxbins_6_Miles_per_Gallon_bins.start, child_Miles_per_Gallon_bin_maxbins_6_Miles_per_Gallon_bins.stop + child_Miles_per_Gallon_bin_maxbins_6_Miles_per_Gallon_bins.step, child_Miles_per_Gallon_bin_maxbins_6_Miles_per_Gallon_bins.step)"
          },
          {
            "signal": "sequence(child_Horsepower_bin_maxbins_6_Horsepower_bins.start, child_Horsepower_bin_maxbins_6_Horsepower_bins.stop + child_Horsepower_bin_maxbins_6_Horsepower_bins.step, child_Horsepower_bin_maxbins_6_Horsepower_bins.step)"
          }
        ],
        "sort": true
      },
      "range": "ramp"
    }
  ]
}
```

Results in

```
vega.js:132 ERROR RangeError: Invalid array length
    at quantize$4 (vega.js:19016)
    at configureScheme (vega.js:19001)
    at configureRange (vega.js:18949)
    at Scale.prototype$58.transform (vega.js:18862)
    at Scale.prototype$7.evaluate (vega.js:4596)
    at Scale.prototype$7.run (vega.js:4577)
    at View.run (vega.js:4112)
    at View.prototype$81.run (vega.js:35492)
    at vega.js:3433
    at View.runAfter (vega.js:4191)
```

**Comment by jheer** [comment]
Thanks! I tracked this down to a subtle operator scheduling issue, which arose due to a missing dependency in the dataflow graph. Fortunately the fix is straightforward and will be included in the next releases of vega-parser and vega.

**Comment by jheer** [comment]
Fixed in vega-parser v2.4.0 and released in vega 3.0.9.
